### PR TITLE
Implement Todo list widget with backend 

### LINKED ISSUE #63: Todo list (Dashboard / Full Screen View) - Connect to backend API
**Is your feature request related to a problem? Please describe.**
The habit tracker is only tested with dummy data and storage. The data need to persist and retrieved from the database.

**Describe the solution you'd like**
Connect the widget with the backend API that will allow data storage and retrieval.

**Describe alternatives you've considered**
Keep dummy data as the default value in case the server failure or loss of network connection.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by cecilia-yin**
Closes #63

### CODE REVIEW

File: telos-frontend/src/components/todo/DashboardTodo.jsx [hunk 1]
```diff
@@ -62,65 +64,38 @@ const outdated = {
...
+  const [todos, setTodos] = useState([]);
   const [anchorEl, setAnchorEl] = useState(null);
   const [cancel, setCancel] = useState([0]);
   const [reDate, setReDate] = useState('');
-  const [selectedTodo, setselectedTodo] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState({});
   const [open, setOpen] = useState(false);
   const [migrate, setMigrate] = useState(false);
 
   function dateToTimestamp(endTime) {
```
Comment (blakehattingh) [inline telos-frontend/src/components/todo/DashboardTodo.jsx:79]: This method is a little confusing. A small comment explaining its purpose may be useful here.

File: telos-frontend/src/components/todo/DashboardTodo.jsx [hunk 2]
```diff
@@ -150,8 +129,7 @@ const DashboardTodo = () => {
     setOpen(false);
   };
 
-  // Adding a todo from the 'New to do' dialog
```
Comment (blakehattingh) [inline telos-frontend/src/components/todo/DashboardTodo.jsx:153]: Was this comment meant to be removed? seems worth keeping.

**Review by blakehattingh — COMMENTED** [review]
This change looks good. The code all looks well written and the changes are as the issue described. One thing is that due to the class being so large with so many methods, it may be worth adding a few explanatory comments to improve its understandability. Perhaps an overall comment at the top of the file would be useful.

File: telos-frontend/src/components/todo/DashboardTodo.jsx (hunk 1 above)
Comment (cecilia-yin) [reply telos-frontend/src/components/todo/DashboardTodo.jsx:79]: This is written by the previous team so I'm not really sure what does it do.

File: telos-frontend/src/components/todo/DashboardTodo.jsx (hunk 2 above)
Comment (cecilia-yin) [reply telos-frontend/src/components/todo/DashboardTodo.jsx:153]: Added that back

**Review by Jaysce — APPROVED** [review]
LGTM

**Review by lilichenyuxin — APPROVED** [review]
LGTM

**Pull request merged**